x-pack/plugins/apm/public/components/app/service_map/Popover/backend_contents.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButton, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ContentsProps } from '.';
+import { NodeStats } from '../../../../../common/service_map';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useApmRouter } from '../../../../hooks/use_apm_router';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { StatsList } from './stats_list';
+
+export function BackendContents({ nodeData }: ContentsProps) {
+  const { query } = useApmParams('/*');
```

Comment:
@dgieselaar 

if I do `useApmParams('/*');`, then when I call `apmRouter.link` below with `query` I get:

```
info x-pack/plugins/apm/public/components/app/service_map/Popover/backend_contents.tsx:53:5 - error TS2322: Type '{} | { pageStep?: AgentConfigurationPageStep | undefined; } | { name?: string | undefined; environment?: string | undefined; pageStep?: AgentConfigurationPageStep | undefined; } | ... 5 more ... | ({ ...; } & { ...; })' is not assignable to type '({ rangeFrom?: string | undefined; rangeTo?: string | undefined; environment?: string | undefined; kuery?: string | undefined; } & { comparisonEnabled?: string | undefined; comparisonType?: string | undefined; }) | undefined'.
        Type '{ pageStep?: AgentConfigurationPageStep | undefined; }' has no properties in common with type '{ rangeFrom?: string | undefined; rangeTo?: string | undefined; environment?: string | undefined; kuery?: string | undefined; } & { comparisonEnabled?: string | undefined; comparisonType?: string | undefined; }'.

      53     query,
             ~~~~~

        x-pack/plugins/apm/public/components/routing/home/index.tsx:57:5
           57     query: t.partial({
                  ~~~~~~~~~~~~~~~~~~
           58       rangeFrom: t.string,
              ~~~~~~~~~~~~~~~~~~~~~~~~~~
          ... 
           61       kuery: t.string,
              ~~~~~~~~~~~~~~~~~~~~~~
           62     }),
              ~~~~~~
          The expected type comes from property 'query' which is declared here on type '{ query?: { rangeFrom?: string | undefined; rangeTo?: string | undefined; environment?: string | undefined; kuery?: string | undefined; } | undefined; } & { query?: { comparisonEnabled?: string | undefined; comparisonType?: string | undefined; } | undefined; } & { ...; }'
```

Since the only think I'm using `query` for is to pass into `link` below should I just do `useApmParams('/backends/:backendName/overview')` here? Or just `as` up the call below?

Reply:
Nevermind I can't do `useApmParams('/backends/:backendName/overview')`, because then I get `no matching route`.

Reply:
...I just used `query as any` below

Reply:
...or `query as TypeOf<
      ApmRoutes,
      '/backends/:backendName/overview'
    >['query'],`